ui/nuxeo-document-picker/nuxeo-document-picker.js

```diff
@@ -0,0 +1,192 @@
+          }
+        </style>
+        <nuxeo-dialog id="dialog" modal>
+          <div id="mainContainer">
+            <div id="topContainer">
+              <iron-icon icon="nuxeo:cross" on-tap="_close"></iron-icon>
+            </div>
+            <nuxeo-results-view
+              id="resultsView"
+              provider="[[provider]]"
+              page-size="[[pageSize]]"
+              params="[[params]]"
+              quick-filters="{{quickFilters}}"
+              schemas="[[schemas]]"
+              enrichers="[[enrichers]]"
+              search-name="[[searchName]]"
+              aggregations="{{aggregations}}"
+              visible
+              show-filters
+              opened
+              search-form="[[searchForm]]"
+              on-navigate="_onNavigate"
+              on-results-changed="_onResultsChanged"
+            ></nuxeo-results-view>
+          </div>
+          <div class="buttons">
+            <!-- TODO check if we can have these stuck to the bottom (absolute positioning? check creation wizard) -->
+            <paper-button noink class="secondary" dialog-dismiss>Cancel</paper-button>
+            <paper-button noink class="primary" on-tap="_onInsert" id="insertButton">Insert</paper-button>
+          </div>
+        </nuxeo-dialog>
+      `;
+    }
+
+    static get is() {
+      return 'nuxeo-document-picker';
+    }
+
+    static get properties() {
+      return {
+        aggregations: Object,
+        enrichers: String,
+        pageSize: Number,
+        params: Object,
+        provider: String,
+        quickFilters: Object,
+        schemas: String,
+        searchForm: Object,
+        searchName: String,
+      };
+    }
+
+    ready() {
+      super.ready();
+      if (!this._nxProvider) {
+        this._nxProvider = this.$.provider;
+      }
+    }
+
+    disconnectedCallback() {
+      super.disconnectedCallback();
+      if (this.listenedResults) {
+        this.listenedResults.removeEventListener('selected-items-changed', this._updateInsertButton.bind(this));
+      }
+    }
+
+    open() {
+      if (this.$.resultsView) {
+        // XXX could make sense to consider having this as a public method on nuxeo-results-view
+        this.$.resultsView._clear();
+      }
+      this._updateInsertButton();
+      this.$.dialog.open();
+    }
+
+    _close() {
+      this.$.dialog.close();
+    }
+
+    _getSelectedItems() {
+      return (
+        this.$.resultsView &&
+        this.$.resultsView.results &&
+        this.$.resultsView.results.results &&
+        this.$.resultsView.results.results.selectedItems
+      );
+    }
+
+    _updateInsertButton() {
+      const selectedItems = this._getSelectedItems();
+      this.$.insertButton.disabled = !(selectedItems && selectedItems.length);
+    }
+
+    _onInsert() {
+      const selectedItems = this._getSelectedItems();
+      if (selectedItems) {
+        this.dispatchEvent(
+          new CustomEvent('inserted', {
```

Comment:
Related to the comment above, "insert" as an event name seems too tied to the context where we are inserting images in an HTML document. If we decide to use this later on for selecting documentos for something else, then it might not make much sense. WDYT?

Reply:
Absolutely.